src/VisualStudio/VisualBasic/Impl/LanguageService/VisualBasicPackage.vb

```diff
@@ -55,9 +55,9 @@ Namespace Microsoft.VisualStudio.LanguageServices.VisualBasic
     <ProvideLanguageEditorOptionPage(GetType(IntelliSenseOptionPage), "Basic", Nothing, "IntelliSense", "#112", 312)>
     <ProvideAutomationProperties("TextEditor", "Basic", Guids.TextManagerPackageString, 103, 105, Guids.VisualBasicPackageIdString)>
     <ProvideAutomationProperties("TextEditor", "Basic-Specific", Guids.VisualBasicPackageIdString, 104, 106)>
-    <ProvideService(GetType(VisualBasicLanguageService), ServiceName:="Visual Basic Language Service")>
-    <ProvideService(GetType(IVbCompilerService), ServiceName:="Visual Basic Project System Shim")>
-    <ProvideService(GetType(IVbTempPECompilerFactory), ServiceName:="Visual Basic TempPE Compiler Factory Service")>
+    <ProvideService(GetType(VisualBasicLanguageService), ServiceName:="Visual Basic Language Service", IsAsyncQueryable:=True)>
```

Comment:
are we sure these are thread-safe? I know Workspace is threadsafe, but those VS services that we assumed to be called from UI thread might not thread-safe.

Reply:
okay, what does it actually mean that a service is IsAysncQueriable?

Reply:
> okay, what does it actually mean that a service is IsAysncQueriable?

@madskristensen can you answer this?

Reply:
I figured out the answer and fixed the implementation in 6aa6b30d8916a0b0a31698e1aa013d220ce35e08.